dev/Lights/MaterialHelper.cpp

```diff
@@ -894,9 +894,12 @@ void MaterialHelper::OnVisibilityChanged(const winrt::CoreWindow&, const winrt::
 // experience severe rendering lag in resize scenarios (Bug 13289165).
 void MaterialHelper::OnSizeChanged(const winrt::IInspectable& /*sender*/, const winrt::IInspectable& /*args*/)
 {
-    const bool isFullScreenOrTabletMode = IsFullScreenOrTabletMode();
-
     const auto strongThis = get_strong();
+
+    // When IsFullScreen changes we get a SizeChanged event.
+    strongThis->m_isFullScreenModeValid = false;
+
+    const bool isFullScreenOrTabletMode = strongThis->IsFullScreenOrTabletModeImpl();
     m_sizeChangedListeners(strongThis, isFullScreenOrTabletMode);
```

Comment:
Nit: Technically we could just pass `strongThis->IsFullScreenOrTabletModeImpl()` to the event and skip saving it in a variable.

Reply:
True, but it was using a local before and I think putting it inline reduces readability slightly. Unfortunately I can't find a CppCoreGuideline rule to support either style.

Reply:
Yes, good point. Since there doesn't seem to be a CppCoreGuideline rule governing this (I haven't found one too) and the CI already passed, I think we should leave it as is and merge the PR. Allocating a bool isn't too bad after all and readability profits from this design.